wallet.md

```diff
@@ -0,0 +1,133 @@
+Wallet API
+==========
+
+BlockAdded Notifications
+------------------------
+```
+type NotifyBlockAddedRequestMessage struct {}
+
+type NotifyBlockAddedResponseMessage struct {
+	Error *RPCError
+}
+
+type BlockAddedNotificationMessage struct {
+	Block *RPCBlock
+}
+```
+ChainChanged Notifications
+--------------------------
+```
+type NotifyChainChangedRequestMessage struct {}
+
+type NotifyChainChangedResponseMessage strcut {
+	Error *RPCError
+}
+
+type ChainChangedNotificationMessage struct {
+	RemovedChainBlockHashes []string
+	AddedChainBlocks []*ChainBlock
+}
+
+type ChainBlock struct {
+	Hash string
+	AcceptedBlocks []*AcceptedBlock
+}
+
+type AcceptedBlock struct {
+	Hash string
+	AcceptedTransactionIDs []string
+}
+```
+UTXOsChanged Notifications
+--------------------------
+```
+type NotifyUTXOsChangedRequestMessage struct {
+	Addresses []string
```

Comment:
Just a thought: Maybe we want the index to be by scriptPublicKey and not addresses? It'll also mean that you won't UTXOsByAddressesEntry won't need to have an address field